### ISSUE TITLE
Message "[Checks API] No suitable checks publisher found." started to appear in pipeline console output

### ISSUE THREAD

**Issue by vladaurosh** [jenkinsci/checks-api-plugin#38]
Hello.
I've updated jenkins plugins today and looks like some of Pipeline plugins installed Checks API Plugin (version 1.0.2), which I didn't have installed.
Now in console output of pipeline jobs I see messages "[Checks API] No suitable checks publisher found." after I use git, and second time at the very end of pipeline. Pipelines work fine, I just get these messages.

Is there any way to suppress these messages, at least as an option?

**Comment by XiongKezhi** [comment]
Which plugin are you using? Currently, there are warnings-ng and code-coverage plugins are using the checks-api plugin, and they all provide configurations to skip publishing checks. E.g, for warning-ng plugins: https://github.com/jenkinsci/warnings-ng-plugin/blob/master/doc/Documentation.md#warnings-checks-for-github-projects

**Comment by vladaurosh** [comment]
Hello.
I do not have any of those 2 plugins. I can provide full list of plugins installed if that would help.
My pipeline is scripted pipeline where I have something like this
stage('SOURCE') {
   git credentialsId: 'xxxxx', branch: "some_branch", url:"some_url"
}
After git command I get "[Checks API] No suitable checks publisher found."

After that stage I have stage where I build app using maven but I am not getting that message in that stage. And at the very end I have this (all stages are finished):

[Checks API] No suitable checks publisher found.
Finished: SUCCESS

**Comment by XiongKezhi** [comment]
Check if you have installed the checks-api-plugin, just uninstall it.

**Comment by vladaurosh** [comment]
I do have it, but it was installed as dependency for some plugins that I updated (not installed) this morning. 
These were updated (or installed as dependency) this morning:
timestamper.jpi
junit.jpi
checks-api.jpi
credentials-binding.jpi
workflow-step-api.jpi


But I can't uninstall it, it is greyed out.

![checks api](https://user-images.githubusercontent.com/47857324/96257888-102ed780-0fb3-11eb-98ed-8478dfb816c3.jpg)

**Comment by vladaurosh** [comment]
Looks like JUnit plugin has Check API plugin as dependency.
https://plugins.jenkins.io/junit/#dependencies

**Comment by XiongKezhi** [comment]
junit plug-in started use this API: https://github.com/jenkinsci/junit-plugin/pull/180

**Comment by vladaurosh** [comment]
Yeah I just saw this:
1.39
Released: 2 days ago
🚀 New features and improvements
Checks API integration (#180) @timja

**Comment by phoenix384** [comment]
This does not only affect console output of pipeline jobs. Every classic freestyle job also has this message in its console output.

**Comment by vladaurosh** [comment]
Hello.
I see that 1.0.3 update is ready, but looks like this change is not there yet?

**Comment by XiongKezhi** [comment]
Yes, that's daily dependencies update. This change will be merged soon, only tests are left.

**Comment by chess-levin** [comment]
We have the same "problem" that junit-plugin prints "[Checks API] No suitable checks publisher found." We are using 1.0.3 of checks-api-plugin.
Is there some progress on this issue?

**Comment by XiongKezhi** [comment]
Just released: https://github.com/jenkinsci/checks-api-plugin/releases/tag/v1.1.0. This should have solved this problem,

**Comment by kad-meedel** [comment]
I am running version 1.2.0 of the checks-api now, but stil get the messages  "[Checks API] No suitable checks publisher found." in the console logs.
Is there any thing else to configure to 'suppress' these messages?

**Comment by XiongKezhi** [comment]
> I am running version 1.2.0 of the checks-api now, but stil get the messages "[Checks API] No suitable checks publisher found." in the console logs.
> Is there any thing else to configure to 'suppress' these messages?

Are you using any consumer plugins of this API (e.g. warnings-ng, junit, or code-coverage)?

**Comment by kad-meedel** [comment]
We have the junit plugin installed, but I am not using this in my jenkins job. This is a 'simple' pipeline job.
The message is shown after a git checkout and at the end of the pipeline job.

**Comment by uhafner** [comment]
Isn't our listener independent of any consumer? That means the message will be written out for every job.

**Comment by XiongKezhi** [comment]
> Isn't our listener independent of any consumer? That means the message will be written out for every job. 

it's independent, but if using a consumer plugin like warnings-ng (without implementation), we'll also see that message. I'm just to make sure it comes from the listener.

**Comment by XiongKezhi** [comment]
@kad-meedel Does you Jenkins install the github checks plugin? I seem to have found a problem, but not sure if that's your case.

**Comment by kad-meedel** [comment]
We do not have a separate install of the checks-api plugin.
I think one of the other plugins has checks-api plugin as an dependencie! Pipeline: step api fore instance.

**Comment by kad-meedel** [comment]
We do have an installation of the GitHub Branch Source pugin. This one has an dependency with github checks

**Comment by XiongKezhi** [comment]
> We do have an installation of the GitHub Branch Source pugin. This one has an dependency with github checks

I may need your double-check for it?

our github checks plugin do dependent on github branch source plugin, but not the other way around.

**Comment by kad-meedel** [comment]
Yes you are right. 
And we do not install the github checks plugin. 
So what would than be the trigger?

**Comment by uhafner** [comment]
https://github.com/jenkinsci/checks-api-plugin/blob/bb070c651870fff42e128af7ce80f18ffde3a870/src/main/java/io/jenkins/plugins/checks/status/BuildStatusChecksPublisher.java#L124

**Comment by XiongKezhi** [comment]
> The message is shown after a git checkout and at the end of the pipeline job.

According to your description, it looks like something goes wrong with our [the status checks feature](https://github.com/jenkinsci/checks-api-plugin#embedded-features) although we have carefully designed it not to publish status checks when no implementation is installed. But I can reproduce it: I deleted the github-checks and runs a template pipeline job, nothing related in its log, however.

```groovy
pipeline {
    agent any

    stages {
        stage('Build') {
            steps {
                // Get some code from a GitHub repository
                git 'https://github.com/jenkins-checks-api-test/codingstyle'

                // Run Maven on a Unix agent.
                sh "mvn -Dmaven.test.failure.ignore=true clean verify"
            }

            post {
                // If Maven was able to run the tests, even if some of the test
                // failed, record the test results and archive the jar file.
                success {
                    archiveArtifacts 'target/*.jar'
                }
            }
        }
    }
}
```

Can you simply [replay](https://www.jenkins.io/doc/book/pipeline/development/#replay) it and show me the log?

**Comment by XiongKezhi** [comment]
> https://github.com/jenkinsci/checks-api-plugin/blob/bb070c651870fff42e128af7ce80f18ffde3a870/src/main/java/io/jenkins/plugins/checks/status/BuildStatusChecksPublisher.java#L124

That's not the problem if not installing the github-checks, since not properties implementation exists at all (other than the default one which is not applicable to any jobs): https://github.com/jenkinsci/checks-api-plugin/blob/bb070c651870fff42e128af7ce80f18ffde3a870/src/main/java/io/jenkins/plugins/checks/status/AbstractStatusChecksProperties.java#L40

**Comment by kad-meedel** [comment]
I have run the pipeline script as you mentioned. I only included the mvn tool in the code. ( is not a default implementation ) 
In the console log output I skipped all the [INFO] Downloaded from InternalRepo: messages, to keep the log small.
[jenkins-checks-testrun.txt](https://github.com/jenkinsci/checks-api-plugin/files/5749910/jenkins-checks-testrun.txt)

**Comment by kad-meedel** [comment]
I have run the pipeline script as you mentioned. I only included the mvn tool in the code. ( is not a default implementation )
[jenkins-checks-testrun.txt](https://github.com/jenkinsci/checks-api-plugin/files/5749931/jenkins-checks-testrun.txt)

**Comment by kad-meedel** [comment]
In a 'simple' pipeline job the message is gone. In a 'simple' jenkins project the message is stil there.

**Comment by mcascone** [comment]
For what it's worth to anyone coming here now, I added `skipPublishingChecks: true` to skip this troublesome part.

I know that our Enterprise GitHub is getting build status updates from Jenkins, so those checks are being published some other way, and I was ok to remove this one.

```java
junit testResults: 'results.xml', skipPublishingChecks: true
```

**Comment by ap200027** [comment]
> Check if you have installed the checks-api-plugin, just uninstall it.

how??
